apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx

```diff
@@ -0,0 +1,255 @@
+import { useCallback, useEffect, useState } from 'react';
+import {
+  Badge,
+  Box,
+  Flex,
+  TextLink,
+  Note,
+  Card,
+  Stack,
+  Paragraph,
+  FormControl,
+  Spinner,
+  Button,
+} from '@contentful/f36-components';
+import { ExternalLinkTrimmedIcon, CheckCircleIcon, ErrorCircleIcon } from '@contentful/f36-icons';
+import { useApi } from 'hooks/useApi';
+import { ServiceAccountKeyId, ServiceAccountKey } from 'types';
+
+interface Props {
+  serviceAccountKeyId: ServiceAccountKeyId;
+  serviceAccountKey: ServiceAccountKey;
+  onEditGoogleAccountDetails: React.MouseEventHandler<HTMLButtonElement>;
+  onAccountSummariesChange: Function;
+}
+
+export const GoogleErrorApiErrorTypes = {
+  AdminAPI: 'AdminApiError',
+  DataAPI: 'DataApiError',
+  Unknown: 'Unknown',
+};
+
+interface GAErrorApiType {
+  details: any;
+  errorType: string;
+  message: string;
+}
+
+const DisplayServiceAccountCard = (props: Props) => {
+  const {
+    serviceAccountKeyId,
+    serviceAccountKey,
+    onEditGoogleAccountDetails,
+    onAccountSummariesChange,
+  } = props;
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [adminApiError, setAdminApiError] = useState<GAErrorApiType>();
+  const [dataApiError, setDataApiError] = useState<GAErrorApiType>();
+  const [unspecifiedAdminError, setUnspecifiedAdminError] = useState<GAErrorApiType>();
+  const [unspecifiedDataError, setUnspecifiedDataError] = useState<GAErrorApiType>();
+
+  // NOTE: Due to a bug installation parameters are not available at sdk.parameters.installation form the config screen
+  // location. Therefore we must pass down the values directly to the useApi hook. If the bug is fixed this won't be
+  // necessary
+  const api = useApi(serviceAccountKeyId, serviceAccountKey);
+
+  const verifyAdminApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const fetchedAccountSummaries = await api.listAccountSummaries();
+      if (fetchedAccountSummaries.res && !fetchedAccountSummaries.res.ok) {
+        setAdminApiError(fetchedAccountSummaries.errors);
+        return;
+      }
+
+      onAccountSummariesChange(fetchedAccountSummaries);
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+    } catch (e: any) {
+      setUnspecifiedAdminError({
+        details: null,
+        errorType: GoogleErrorApiErrorTypes.AdminAPI,
+        message: 'Unknown Google Admin API Error',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+
+    return () => {
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+      setIsLoading(false);
+    }
+  
+  // It wants to add onAccountSummariesChange as a dependency but this will cause an infinite re-render
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [api]);
+
+  const verifyDataApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const pagedData = await api.getPageData();
```

Comment:
stupid naming nitpick but these api names are a bit weird since `getPageData()` and especially `pagedData` really make me think of pagination.

Reply:
i think `paged` is a typo? `pageData` would be fine to me

Reply:
- [ ] fix typo and consider better name (overall) than `pageData` – perhaps `pageviewData` or something aligned with GA's terms

Reply:
Ugh yeah @ryunsong-contentful and i discussed the naming thing a bit yesterday.

In general I think it's nice to not to do arbitrary transformation of names. So if you do `getFoobar` then it's usually a `foobar` that you get.

I also think the lambda is largely acting as a proxy, so it's easier to reason about it if we not only don't do arbitrary transformation of the data responses we get, but leave the API endpoint names alone as well.

In this case we're calling and endpoint in Google:  `POST  https://analyticsdata.googleapis.com/v1beta/properties/GA4_PROPERTY_ID:runReport` ... and that's where things get sucky.

Cuz they names this vaguely restful API something super RPC and it's just annoying and hard to reason about.

Still, I'd rather use a name here that tells the developer what actual endpoint it refers to so they don't have to follow a long and arduous path through our code base to figure out what data is called and returned.

I don't have any great ideas. I will say that `runReport` actually _returns_ a `ReportResponse` object: https://developers.google.com/analytics/devguides/reporting/data/v1/basics#report_response . I mean maybe it's just `getReport()`? 

Anyhow I don't have a super strong opinions here except to guide us toward transparent naming vs. arbitrary renaming.